lib/archethic_web/live/chains/node_shared_secrets_live.ex

```diff
@@ -0,0 +1,221 @@
+
+    updated_socket =
+      case current_page do
+        1 when display_txs < @display_limit ->
+          nb_authorized_nodes = nb_of_authorized_keys(address)
+
+          socket
+          |> update(:transactions, &[display_data(address, nb_authorized_nodes, timestamp) | &1])
+          |> assign(:tx_count, tx_count + 1)
+          |> assign(:nb_authorized_nodes, nb_authorized_nodes)
+
+        1 when display_txs >= @display_limit ->
+          nb_authorized_nodes = nb_of_authorized_keys(address)
+
+          socket
+          |> assign(:tx_count, tx_count + 1)
+          |> assign(:current_page, 1)
+          |> assign(:nb_authorized_nodes, nb_authorized_nodes)
+          |> assign(:nb_pages, total_pages(tx_count + 1))
+          |> assign(:transactions, [display_data(address, nb_authorized_nodes, timestamp)])
+
+        _ ->
+          socket
+      end
+
+    {:noreply, updated_socket}
+  end
+
+  @spec transactions_from_page(current_page :: non_neg_integer(), tx_count :: non_neg_integer()) ::
+          list(map())
+  def transactions_from_page(current_page, tx_count) do
+    nb_drops = tx_count - current_page * @display_limit
+
+    {nb_drops, display_limit} =
+      if nb_drops < 0, do: {0, @display_limit + nb_drops}, else: {nb_drops, @display_limit}
+
+    case SharedSecrets.genesis_address(@txn_type) do
+      address when is_binary(address) ->
+        address
+        |> TransactionChain.list_chain_addresses()
+        |> Stream.drop(nb_drops)
+        |> Stream.take(display_limit)
+        |> Stream.map(fn {addr, timestamp} ->
+          nb_authorized_nodes = nb_of_authorized_keys(addr)
+
+          display_data(
+            addr,
+            nb_authorized_nodes,
+            DateTime.from_unix(timestamp, :millisecond) |> elem(1)
+          )
+        end)
+        |> Enum.reverse()
+
+      _ ->
+        []
+    end
+  end
+
+  @spec nb_of_authorized_keys(address :: binary()) :: non_neg_integer()
+  defp nb_of_authorized_keys(address) do
+    TransactionChain.get_transaction(address, data: [:ownerships])
+    |> elem(1)
+    |> get_in([Access.key(:data), Access.key(:ownerships)])
+    |> Enum.at(0)
+    |> get_in([Access.key(:authorized_keys)])
+    |> Enum.count()
+  end
+
+  @spec display_data(
+          address :: binary(),
+          nb_authorized_nodes :: non_neg_integer(),
+          timestamp :: DateTime.t()
+        ) ::
+          map()
+  defp display_data(address, nb_authorized_nodes, timestamp) do
+    %{
+      address: address,
+      type: @txn_type,
+      timestamp: timestamp,
+      nb_authorized_nodes: nb_authorized_nodes
+    }
+  end
+
+  @doc """
+   Nb of pages required to display all the transactions.
+
+   ## Examples
+      iex> total_pages(45)
+      5
+      iex> total_pages(40)
+      4
+      iex> total_pages(1)
+      1
+      iex> total_pages(10)
+      1
+      iex> total_pages(11)
+      2
+      iex> total_pages(0)
+      0
+  """
+  @spec total_pages(tx_count :: non_neg_integer()) ::
```

Comment:
This function is used in many files of the explorer, maybe you can put it in utils file

Reply:
It would make sense to create a Util module in archethic_web